### PR TITLE
Speed up CodeQL job

### PR DESCRIPTION
**Pull request by sunjayBhatia**
- Use go action cache
- Select only Go as language
- Remove HEAD^2 checkout, prevents warning in codeql init step

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -22,33 +22,27 @@ jobs:
...
+          ${{ runner.os }}-${{ github.job }}-go-
 
     # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v1
       # Override language selection by uncommenting this and choosing your languages
-      # with:
-      #   languages: go, javascript, csharp, python, cpp, java
+      with:
+        languages: go
```
Comment (sunjayBhatia) [inline .github/workflows/codeql-analysis.yml:41]: I'm assuming we only care about this, previously it was also scanning javascript from our site, not sure if we want to add that back

**Review by skriss — APPROVED** [review]
LGTM

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (youngnick) [reply .github/workflows/codeql-analysis.yml:37]: I think this is fine for now.

**Review by youngnick — APPROVED** [review]
LGTM, nice work.

**Pull request merged**